docs/book/src/roadmap.md

```diff
@@ -0,0 +1,42 @@
+# Cluster API Roadmap
+
+This roadmap is a constant work in progress, subject to frequent revision. Dates are approximations.
+
+## Ongoing
+
+- Documentation improvements
+
+## v0.3 (v1alpha3) ~ March 2020
+
+- Control plane provider concept, KubeadmControlPlane implementation
+- clusterctl v2 - easier installation & upgrades of Cluster API + providers
+- Integration with cloud provider VM scaling groups (MachinePools)
+- Machine failure remediation
+- Provider end to end testing framework
+
+## v0.4 (v1alpha4) ~ July 2020
+
+- [Machine bootstrap failure detection](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/972)
+- [Improved status conditions](https://github.com/kubernetes-sigs/cluster-api/issues/1658)
+- [Machine load balancers](https://github.com/kubernetes-sigs/cluster-api/issues/1250)
+- [Extensible machine pre-delete hooks](https://github.com/kubernetes-sigs/cluster-api/issues/1514)
+
+## v0.5 (v1???) ~ November 2020
+
+- ?
+
+## TBD
+
+- [Pluggable MachineDeployment upgrade strategies](https://github.com/kubernetes-sigs/cluster-api/issues/1754)
+- Dual stack IPv4/IPv6 support
+- Windows nodes
+- Auto-scaling support (either with cluster-autoscaler, or something else)
+- UEFI booting
```

Comment:
Is this something that falls under core cluster api or under specific machine infrastructure providers that may support alternative methods for booting?

Reply:
I'm not sure 😄. It came from previous brainstorming sessions. @randomvariable do you remember?

Reply:
Image Builder needs UEFI/BIOS dual support:
* CAPV requires UEFI to be able to leverage the vTPM.
* Azure also requires UEFI boot for Generation 2 VMs

Dual-stack IPv4/IPv6 is mostly about testing - do we rely on single IPs anywhere, kubeadm etc...

Windows support is image-builder, plus anything that needs to be done for CAPI (if XML Answer files are to be supported)

Cluster autoscaling will need to be split across CAPI and providers.